### ISSUE TITLE
Trouble adding labels in Jupyter notebook

### ISSUE THREAD

**Issue by ldclaar** [BrancoLab/BrainRender#70]
Hello,
Thanks for building this tool! I am using BrainRender in a Jupyter notebook to visualize some cortical regions (fairly simple). I would like to add labels to the regions, but it's not working.

**Describe the bug**
Using the Scene.add_brain_regions(['MOs'], add_labels=True) and Scene.add_text('title') in Jupyter notebook does not add labels (when hovering) nor "title" text on the rendered scene.

**To Reproduce**
Windows 10; Python 3.7.7; Jupyter 6.0.3; brainrender 0.3.4.4b0; vtkplotter 2020.3.0

I'm using the following code:
`import vtkplotter as vtk
from brainrender.scene import Scene

scene = Scene(jupyter=True)
scene.add_brain_regions(['MOp', 'MOs', 'VISl', 'VISp', 'RSPagl'], add_labels=True)
scene.add_text('title')
scene.render()

vp = vtk.Plotter()
vp.show(scene.get_actors())`

![upside_down_BR](https://user-images.githubusercontent.com/45570217/81965983-ad817c00-95cd-11ea-9454-5b1258d4ba16.PNG)
No printed errors. It renders the scene, but when I hover over the brain regions no labels are visible. Also, the "title" from the .add_text() function is not visible.

**Expected behavior**
Expected to see brain region labels when hovering over them and a "title" to appear on the scene.

* Side note: Is there a way to get the brain to show at a different angle? It always appears upside-down. I can re-orient it manually, but I thought there might be a setting in one of the functions.

**Comment by FedeClaudi** [comment]
Hi, 

Thanks for getting in touch and for the detailed description. 

The short answer is that embedding of scenes in jupyter notebooks is a bit experimental still and should be avoided whenever possible. 
By default `brainrender` sets things up so that `vtkplotter` relies on `k3d` to embed the scene nicely, but this means that some features will be missing (e.g. orienting the camera view and adding a title to the scene). 

If you want to use jupyter notebooks nonetheless, the easiest workaround is to avoid using `k3d`. This can be done with two small changes to your code:
```
from vtkplotter import embedWindow
from brainrender.scene import Scene

# Change 1 Disable scene embedding with k3d in vtkplotter
embedWindow(False) 

# Change 2 set jupyter to False when creating the scene
scene = Scene(jupyter=False)

# Populate and render 
scene.add_brain_regions(['MOp', 'MOs', 'VISl', 'VISp', 'RSPagl'], add_labels=True)
scene.add_text('title')
scene.render(camera='top')
```

This will make a vtk window pop up with your scene rendered correctly, with title camera angle and all. All of brainrender's functionality will then be working correctly. 


I hope this helps, but do let me know if anything is unclear. 





P.s.: I would avoid `import vtkplotter as vtk` just because there's another package called `vtk` so it might be confusing if you do that.

**Comment by marcomusy** [comment]
@FedeClaudi technical note: maybe you can drop change nr.2 if you pick the False from `vtkplotter.settings.notebookBackend`

**Comment by ldclaar** [comment]
Thanks for your help! The changes you suggested resulted in a pop-up window with the labels and title. Just curious, is there a feature that makes the labels permanently visible over each brain region?

Also, thanks for the tip about not importing as vtk!

**Comment by FedeClaudi** [comment]
Thanks Marco that's good advice. 

@ldclaar  that wasn't possible, but I've just added it in. 

You'll have to upgrade your version of brainrender from the github repo with:
```
    pip install -U git+https://github.com/BrancoLab/BrainRender.git
```

Then you can check this example to see how to add labels:
https://github.com/BrancoLab/BrainRender/blob/master/Examples/basic/adding_labels.py

You should get something like this:
<img width="1680" alt="Screenshot 2020-05-15 at 10 20 48" src="https://user-images.githubusercontent.com/17436313/82034196-c5bcce00-9695-11ea-9041-decb5dfb7055.png">

The text will follow your camera so it will always be facing you as you move the scene around, but given that it's still an actor in the scene, if there's something else in front of it it will be occluded.

Hope this helps.

**Comment by ldclaar** [comment]
That does help, thanks! What determines the location of the label relative to the brain region? For instance, I am plotting both MOs and MOp and the labels obscure each other. For other regions the label shows up on the other hemisphere. Is it possible to add an argument to the add_actor_label function that offsets the label some distance?

**Comment by FedeClaudi** [comment]
They are currently appearing on the highest point on the mesh, so for MOs and MOp they will be nearby. Maybe a better alternative would be to have them appear over the center of mass of the mesh and shifted up by a little bit. I will add that in and add user inputs to shift label location.

**Comment by FedeClaudi** [comment]
These issues have been fixed with 600565f4c77b9075880d163fe0748ec5c2f1493a

Changes:

- Labels appear on the right hemisphere by default
- added `xoffset`, `yoffset` and `zoffset` to manually alter the location of the label
- added a colored sphere to the point on the mesh that corresponds to the label location [optional]

To try the updated code, you can update `brainrender` from the Github repo with:
```
https://github.com/BrancoLab/BrainRender.git
```

Hope this helps, let me know if you find any problem with this or if you think of ways to improve on this functionality. 

Thanks, 
Federico

**Comment by ldclaar** [comment]
Thanks for all of your help, @FedeClaudi!
